### PR TITLE
Input/Output of Statmech Calculations

### PR DESCRIPTION
**Pull request by ndharms**
When creating Arkane input files for species and transition states, we would first check to make sure that an Arkane input file was created and if it was created, we would not create a new one on the assumption that this was correct. This PR allows for new files to be written and if an old file exists, it will be written as a backup file in case a user wants to keep it.

### CODE REVIEW

**Comment by davidfarinajr** [comment]
@nateharms, can you rebase this branch onto master? I think there are some commits here we already merged in

**Comment by ndharms** [comment]
Okay, I rebased

**Comment by ndharms** [comment]
Because this PR is a little old, here's a summary of the changes:

`calculator/statmech.py`
* Added a few imports
* Added former rotor treatment for stable species
* Added a try / except for running kinetics with rotors and then without if Arkane fails

`species.py`
* reorganized when certain attributes are set

`data/inputoutput.py`
* added `TemplateReaction` to the `read_kinetics_file` method

`job/job.py`
* removed an import

`environment.yml`
* added a `matplotlib` requirement to fix errors that we currently have

**Comment by ndharms** [comment]
@skrsna could you give this one a look over too?

File: autotst/calculator/statmech.py [hunk 1]
```diff
@@ -176,33 +187,42 @@ def get_rotor_info(self, conformer, torsion_index):
...
             )
+            label = conformer.smiles + f'_36by10_{j}_{k}'
 
         if not os.path.exists(tor_log):
             logging.info(
                 f"Torsion log file does not exist for {torsion}")
             return ""
+        try:
+            validated = all(Job(directory=self.directory).verify_rotor(conformer=conformer, label=label))
+        except:
```
Comment (skrsna) [inline autotst/calculator/statmech.py]: maybe add the exact Error you get from this `try/except` block. Plain `except` will catch everything (even an KeyboardInterupt)

File: autotst/calculator/statmech.py [hunk 2]
```diff
@@ -502,10 +542,16 @@ def run(self):
...
 
-        for job in self.kinetics_job.job_list:
-            if isinstance(job, arkane.main.KineticsJob):
+        try:
+            for job in self.kinetics_job.job_list:
+                if isinstance(job, arkane.kinetics.KineticsJob):
+                    self.kinetics_job = job
+                elif isinstance(job, arkane.thermo.ThermoJob):
+                    self.thermo_job = job
+        except:
```
Comment (skrsna) [inline autotst/calculator/statmech.py]: same here. Try to catch the exact Errors and maybe log them as well

File: autotst/calculator/statmech.py [hunk 3]
```diff
@@ -106,8 +111,12 @@ def write_conformer_file(self, conformer):
             return False
 
         if os.path.exists(os.path.join(self.directory, "species", label, label + '.py')):
```
Comment (skrsna) [inline autotst/calculator/statmech.py:113]: maybe move the `old_path` or rename it to PATH and check it 
e.g. 
```
PATH = os.path.join(self.directory, "species", label, label + '.py'
if os.path.exists(PATH):
   logging.info(f"Species input file already written... Renaming it {old_path} and creating a new one.")
             shutil.move(
                 PATH,
                 PATH.replace('py', 'old.py')
             )
```

File: autotst/calculator/statmech.py [hunk 4]
```diff
@@ -231,9 +251,13 @@ def write_ts_input(self, transitionstate):
         label = transitionstate.reaction_label
 
         if os.path.exists(os.path.join(self.directory, "ts", label, label + '.py')):
```
Comment (skrsna) [inline autotst/calculator/statmech.py:253]: same comment as above :)

**Review by skrsna — CHANGES_REQUESTED** [review]

**Comment by ndharms** [comment]
@skrsna the corrections that you requested were made. When you get a chance, can you approve this PR?

**Review by skrsna — APPROVED** [review]

**Pull request merged**